sky/backends/backend_utils.py

```diff
@@ -1450,16 +1450,14 @@ def _get_tpu_vm_pod_ips(ray_config: Dict[str, Any],
                          'Hint: make sure it is not leaked.')
             continue
 
-        if not get_internal_ips:
-            ips = [
-                endpoint['accessConfig']['externalIp']
-                for endpoint in tpuvm_json['networkEndpoints']
-            ]
-        else:
-            ips = [
-                endpoint['ipAddress']
-                for endpoint in tpuvm_json['networkEndpoints']
-            ]
+        ips = []
+        for endpoint in tpuvm_json['networkEndpoints']:
+            # Note: if TPU VM is being preempted, its IP field may not exist.
+            # We use get() to avoid KeyError.
+            ip = (endpoint.get('ipAddress', None) if get_internal_ips else
+                  endpoint['accessConfig'].get('externalIp', None))
```

Comment:
nit: having the expanded `if...else...` may be easier to read?

Reply:
Fixed!